core/psh/login.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Phoenix-RTOS
+ *
+ * login - allows/disallows access to Phoenix SHell 
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "psh.h"
+
+
+int psh_login(void){
+	char c, *usrpasswd;
+	unsigned int passsz = 0;
+
+	/* allocate memory for user-given password */
+	if ((usrpasswd = calloc(strlen(psh_common.passwd),sizeof(char)+1)) == NULL )
+	{
+		fprintf(stderr,"psh: fail to allocate memory for login");
+		return -ENOMEM;
+	}
+
+	for (;;)
+	{
+		if( !passsz ){
```

Comment:
```suggestion
		if (!passsz ) {
```
Those unnecessary spaces seem to appear in many places, so I'm not going to mark all of them.
What's more, I think that, since we are thinking of this variable as an integer, it would be more readable to compare it to 0: `passsz == 0`.

Reply:
logic error: if I input 1 char and then backspace, the `Enter password: ` string would be printed again

Reply:
@MaciejPurski about spaces: i am not going to overuse them again. Thanks. About logic statement - shorter version does have impact on code readability. Going to use longer version

@nalajcie yup - going to implement "one way escape" from "Enter password" code block